pkg/model/api.go

```diff
@@ -51,6 +56,97 @@ func ImportModel(ctx context.Context, filePath, projectId, message, modelType, b
 		return err
 	}
 
-	log.Println("Starting import model job. JobId: ", importModelData.GetImportModelJobId())
+	importModelJobId := importModelData.GetImportModelJobId()
+	if waitForResults {
+		okStatus, job, err := waitForImportModelJob(ctx, importModelJobId)
+		if err != nil {
+			return err
+		}
+		if okStatus {
+			log.Println("Successfully imported model")
+			if mappingYaml != "" {
+				printMappingValidationErrors(ctx, job.VersionId, projectId, mappingYaml)
+			}
+		} else {
+			log.Println("Failed to import model")
+		}
+		return nil
+	}
+
+	log.Println("Starting import model job. JobId: ", importModelJobId)
+	return nil
+}
+
+func waitForImportModelJob(ctx context.Context, importModelJobId string) (ok bool, job *tlApi.RunProcess, err error) {
+	message := "Waiting for import model result..."
+	sleepDuration := 3 * time.Second
+	timeoutDuration := 10 * time.Minute
+
+	condition := func() (bool, error) {
+		data, _, err := api.ApiClient.GetTeamJobs(ctx).Execute()
+		if err != nil {
+			return false, fmt.Errorf("failed to wait for the import model job status: %v", err)
+		}
+		if len(data.Processes) == 0 {
+			return false, fmt.Errorf("failed to wait for the import model job status")
+		}
+
+		job = findRunProcessByJobId(data.Processes, importModelJobId)
+		switch job.Status {
+		case tlApi.JOBSTATUS_FAILED:
+			ok = true
+			return true, nil
+		case tlApi.JOBSTATUS_FINISHED:
+			ok = true
+			return true, nil
+		}
+
+		return false, nil
+	}
+
+	err = api.WaitForCondition(ctx, message, condition, sleepDuration, timeoutDuration)
+
+	if err == api.ErrorTimeout {
+		return false, nil, fmt.Errorf("timeout occurred while waiting for import job status")
+	}
+	if err != nil {
+		return false, nil, fmt.Errorf("failed to wait for import job result status: %v", err)
+	}
+
+	return
+}
+
+func findRunProcessByJobId(runProcesses []tlApi.RunProcess, jobId string) *tlApi.RunProcess {
+	for _, rp := range runProcesses {
+		if rp.JobId == jobId {
+			return &rp
+		}
+	}
 	return nil
 }
+
+func printMappingValidationErrors(ctx context.Context, versionId string, projectId string, mapping string) {
+	getValidationErrorsParams := *tlApi.NewCodeIntegrationMappingErrorsParams(projectId, versionId, mapping)
+
+	data, _, err := api.ApiClient.GetCodeIntegrationMappingErrorsByVersionId(ctx).CodeIntegrationMappingErrorsParams(getValidationErrorsParams).Execute()
+	if err != nil {
+		log.Println("failed getting mapping validation errors:", err)
+		return
+	}
```

Comment:
Consider printing some contect of the meaning of the table (e.g. We found the following errors in your current mapping file)

Reply:
I do add it right before the print:
encountered the following validation errors while applying mapping to graph